<?xml version="1.0" ?>
<launch>
    <!-- Create gazebo world -->
    <include file="$(find gazebo_ros)launch/empty_world.launch" >
        <arg name="world_name" value="$(find turtlebot3_gazebo)/worlds/turtlebot3_house.world" />
        <arg name="gui" value="false" />
        <arg name="use_sim_time" value="true" />
        <arg name="paused" value="false" />
    </include>
    
    <!-- spawning turtlebot -->
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_burger_for_autorace.urdf.xacro" />
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_burger -x 0.5 -y 0.5 -z 0 -Y 0 -param robot_description" />

	<include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch"/>

    <!-- laden der aufgenommenen map in rviz -->
    <!--<node pkg="map_server" name="map_server" type="map_server" args="$(find problab_goebel)/maps/mymap.yaml"/>-->
    <node pkg="map_server" name="map_server" type="map_server" args="$(find problab_goebel)/maps/kleineMap.yaml"/>

    <!-- laden des roboters in die map-->
    <!-- laut Doku: http://wiki.ros.org/tf#static_transform_publisher -->
    <!-- static_transform_publisher args: x y z qx qy qz qw frame_id child_frame_id  period_in_ms -->
    <node pkg="tf" type="static_transform_publisher" name="tf_map_odom" args="0 0 0 0 0 0 1 map odom 100"/>

    <!-- amcl für vergleich -->
<!--
   <node pkg="amcl" type="amcl" name="amcl">
    <param name="min_particles"             value="500"/>
    <param name="max_particles"             value="3000"/>
    <param name="kld_err"                   value="0.02"/>
    <param name="update_min_d"              value="0.20"/>
    <param name="update_min_a"              value="0.20"/>
    <param name="resample_interval"         value="1"/>
    <param name="transform_tolerance"       value="0.5"/>
    <param name="recovery_alpha_slow"       value="0.00"/>
    <param name="recovery_alpha_fast"       value="0.00"/>
    <param name="initial_pose_x"            value="0.5"/>
    <param name="initial_pose_y"            value="0.5"/>
    <param name="initial_pose_a"            value="0"/>
    <param name="gui_publish_rate"          value="50.0"/>

    <remap from="scan"                      to="/scan"/>
    <param name="laser_max_range"           value="3.5"/>
    <param name="laser_max_beams"           value="180"/>
    <param name="laser_z_hit"               value="0.5"/>
    <param name="laser_z_short"             value="0.05"/>
    <param name="laser_z_max"               value="0.05"/>
    <param name="laser_z_rand"              value="0.5"/>
    <param name="laser_sigma_hit"           value="0.2"/>
    <param name="laser_lambda_short"        value="0.1"/>
    <param name="laser_likelihood_max_dist" value="2.0"/>
    <param name="laser_model_type"          value="likelihood_field"/>

    <param name="odom_model_type"           value="diff"/>
    <param name="odom_alpha1"               value="0.1"/>
    <param name="odom_alpha2"               value="0.1"/>
    <param name="odom_alpha3"               value="0.1"/>
    <param name="odom_alpha4"               value="0.1"/>
    <param name="odom_frame_id"             value="odom"/>
    <param name="base_frame_id"             value="base_footprint"/>
  </node>
-->

    <include file="$(find problab_goebel)/launch/move_base.launch"/>

    <!-- <node type="rviz" name="rviz" pkg="rviz"/> -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find problab_goebel)/config/rviz_config.rviz"/>

    <!-- übergeben der hyperparameter aus dem .yaml -->
    <rosparam file="$(find problab_goebel)/param/hyperparameter.yaml" />

    <!-- eigener particle filter -->
    <node type="particle_filter" name="particle_filter" pkg="problab_goebel" />

    
</launch>
